### PR TITLE
ICU4C Updating Number Format

### PR DESCRIPTION
**Pull request by sven-oly**

### CODE REVIEW

File: executors/cpp/number_fmt.cpp [hunk 1]
```diff
@@ -48,38 +52,248 @@ using std::endl;
...
+
+  string notation_string = "";
+  string precision_string = "";
+  string unit_string = "";
+  string unitDisplay_string = "";
+  string style_string = "";
+  string currency_string = "";
+  string roundingMode_string = "";
+  string compactDisplay_string = "";
+  string conformahceScale_string = "";
```
Comment (sffc) [inline executors/cpp/number_fmt.cpp:143]: conformahceScale_string?

File: executors/cpp/number_fmt.cpp [hunk 2]
```diff
@@ -91,35 +305,59 @@ const string test_numfmt(json_object *json_in) {
...
+  else {
+    // Use settings to initialize the formatter
+    nf = NumberFormatter::withLocale(displayLocale)
+         .notation(notation_setting)
+         .precision(precision_setting)
+         .integerWidth(integerWidth_setting)
+         .scale(scale_setting)
+         .unit(unit_setting)
+         .unitWidth(unit_width_setting);
+  }
```
Comment (sffc) [inline executors/cpp/number_fmt.cpp:334]: To generalize, put the common stuff outside the if statement and then do `nf = nf.unit(...)` inside the if

File: executors/cpp/number_fmt.cpp [hunk 3]
```diff
@@ -91,35 +305,59 @@ const string test_numfmt(json_object *json_in) {
...
 
-  // TEMPORARY - just return the input.
-  test_result = input_string;
+  LocalizedNumberFormatter nf;
+  if (notation_string == "scientific") {
+    notation_setting = Notation::scientific();
+  }
 
-  // TODO!!!
-  //  SimpleNumberFormatter snf = SimpleNumberFormatter::forLocale(displayLocale, status);
```
Comment (sffc) [inline executors/cpp/number_fmt.cpp:100]: This approach is fine but I would also have been okay with an approach that constructs a NumberFormatter from the skeleton string. If we don't use the skeleton string, I think we should remove it from the input data and keep it, at best, in some sort of annotation.

File: executors/cpp/number_fmt.cpp (hunk 2 above)
Comment (sven-oly) [reply executors/cpp/number_fmt.cpp]: I've revised this to merge the two styles.

File: executors/cpp/number_fmt.cpp (hunk 3 above)
Comment (sven-oly) [reply executors/cpp/number_fmt.cpp:100]: Now working on using the skeleton where it exists.

**Review by echeran — APPROVED** [review]
rslgtm

@sven-oly is going to follow up with another PR to address the remaining errors. I'll take a closer look at the code then. But this PR that brings it down from 75% errors to 20% errors is a good first step.

**Pull request merged**